### PR TITLE
endpoint: Update proxy policy after adding redirects

### PR DESCRIPTION
**Pull request by jrajahalme**
Proxy policy must be updated after redirects have been added, as
waiting for ACKs for the current version from the proxy is disabled
when there are no redirects (yet). This leads to the Cilium API
updating the realized policy too soon, and may lead to test flakes due
to traffic being denied because policy has not been initialized yet.

Found in local testing with dev VM proxy smoke test.
```release-note
Changed to update proxy policy after listeners are created to avoid a rare flake of dropping proxy traffic before policy is configured.
```

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by jrajahalme** [comment]
Removed 1.6 backport label as the bug is not causing security failing open.

**Review by christarazi — APPROVED** [review]

File: tests/envoy-smoke-test.sh [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    "ingress": [{
+        "fromEndpoints": [
+	    {"matchLabels":{"reserved:host":""}},
+	    {"matchLabels":{"id.client":""}}
+	]
+    }]
+},{
+    "endpointSelector": {"matchLabels":{"id.client":""}},
+    "egress": [{
+        "fromEndpoints": [{"matchLabels":{"id.server":""}}],
```
Comment (fristonio) [inline tests/envoy-smoke-test.sh]: Should this be `toEndpoints` since this is an egress rule?

File: tests/envoy-smoke-test.sh [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+TEST_NAME=$(get_filename_without_extension $0)
+LOGS_DIR="${dir}/cilium-files/${TEST_NAME}/logs"
+redirect_debug_logs ${LOGS_DIR}
+
+set -ex
+
+function cleanup {
+  monitor_stop
+  echo "Tip: Add '--debug-verbose=flow' into cilium options in /etc/sysconfig/cilium to get Envoy debug logging."
+# These are commented to allow for easier debugging, but are left as comments to remind how to clean up.
```
Comment (fristonio) [inline tests/envoy-smoke-test.sh]: Should we have these comments at the same indentation level as the function body?

**Review by fristonio — CHANGES_REQUESTED** [review]
Looks good mostly, some questions/feedback.
:100:

File: tests/envoy-smoke-test.sh (hunk 1 above)
Comment (jrajahalme) [reply tests/envoy-smoke-test.sh]: Yes! Due to this error the policy has applied to all endpoints, but the test only tests to 'server' so the test did not fail.

File: tests/envoy-smoke-test.sh (hunk 2 above)
Comment (jrajahalme) [reply tests/envoy-smoke-test.sh]: Fixed.

**Comment by jrajahalme** [comment]
test-me-please

**Review by fristonio — APPROVED** [review]
LGTM :rocket:

**Comment by aanm** [comment]
retest-4.9

**Comment by aanm** [comment]
retest-runtime

**Comment by aanm** [comment]
retest-gke

**Pull request merged**